Sources/FoundationEssentials/JSON/JSONDecoder.swift

```diff
@@ -1128,6 +1128,11 @@ extension JSONDecoderImpl : SingleValueDecodingContainer {
     func decode(_: Int64.Type) throws -> Int64 {
         try decodeFixedWidthInteger()
     }
+  
+    @available(macOS 15.0, iOS 18.0, tvOS 18.0, watchOS 11.0, visionOS 2.0, *)
```

Comment:
In other places in the project, we use availability like `FoundationPreview 0.4` (which "back deploys" to macOS 13.3) or `FoundationPredicateRegex 0.4` (which has macOS 15 aligned availability). Is it worth using/creating a FoundationPreview-based availability specifier for this, or should we just explicitly write out the availability here like it is now?

Reply:
Another benefit of using the FoundationPreview macro is that it is defined to be the min deployment target when building the package, making it possible to test this implementation on older OSes instead of requiring a host OS of macOS 15.

Reply:
I think in this case we likely do need a host OS of macOS 15 since I don't think `UInt128` will be present on macOS <15, right?

Reply:
I see, that's why you mentioned the Predicate-specific availability here. I was hoping those wouldn't proliferate too much, but maybe we don't have a choice...

Reply:
Yeah, we need to match the availability of Int128, so I think it's fine to write these out explicitly.